### PR TITLE
perforce: package p4 CLI in gitserver

### PR DESCRIPTION
**Pull request by unknwon**
Install p4 CLI in our gitserver Docker image, this is required for communicating with Perforce Server.

Fixes https://github.com/sourcegraph/sourcegraph/issues/17596

### CODE REVIEW

**Review by tsenart — APPROVED** [review]
LGTM. @pecigonzalo should take a look too.

**Comment by pecigonzalo** [comment]
@unknwon LGTM but I would suggest we move this to be multi-layer. If you need help feel free to ask in #distributioneers

**Comment by unknwon** [comment]
@pecigonzalo Thanks for the suggestion! I changed it to be multi-layer, PTAL!

**Comment by daxmc99** [comment]
After pairing with @unknwon this seems to be the only working solution. 
Other options such as adding `gcompat` and `libc6-compat` didn't work for this binary

**Comment by pecigonzalo** [comment]
LGTM

File: cmd/gitserver/Dockerfile [hunk 1]
```diff
@@ -18,7 +26,17 @@ LABEL com.sourcegraph.github.url=https://github.com/sourcegraph/sourcegraph/comm
...
-    openssh-client
+    openssh-client \
+    git-p4
+
+# hadolint ignore=DL3022
+COPY --from=p4cli /usr/local/bin/p4 /usr/local/bin/p4
+
+# This is a trick to include libraries required by p4,
+# please refer to https://blog.tilander.org/docker-perforce/
+ADD https://github.com/jtilander/p4d/raw/master/lib/lib-x64.tgz /
```
Comment (unknwon) [inline cmd/gitserver/Dockerfile:37]: @pecigonzalo: @daxmc99 and I couldn't figure out which relative path it is to put this `lib-x64.tgz` in our main repository to be able to use `ADD lib-x64.tgz /` (with a local path), so this ended up in fetching from a network address.

I don't feel comfortable with the current approach, do you have recommendation where to put the `lib-x64.tgz` and we are still able to include it while building our Docker images? Thanks!

**Comment by unknwon** [comment]
Merging ahead to catch 3.25 branch cut, I've tested the single-docker-image end-to-end that works for cloning Perforce depots. Happy to address any post-merge feedback!

File: cmd/gitserver/Dockerfile (hunk 1 above)
Comment (pecigonzalo) [reply cmd/gitserver/Dockerfile:37]: @unknwon I think its fine fetching from a URL with ADD, the main concern is that the URL is `master` base. What is your concern?

Comment (unknwon) [reply cmd/gitserver/Dockerfile:37]: This feels like fetching executables from a third-party source and would have problem when we do security auditing?

Comment (pecigonzalo) [reply cmd/gitserver/Dockerfile:37]: Its an `https` link, so its somewhat better, but in any case, we are adding it straight from the URL anyway.

**Pull request merged**